### PR TITLE
Products and categories API tests

### PR DESCRIPTION
**Pull request by ajlasisic**
API tests for Product and Categories APIs using Axios library

### CODE REVIEW

File: test/specs/test.e2e.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  it("Check status code", async () => {
+    await axios
+      .get("http://localhost:8080/api/categories")
+      .then(function (response) {
+        expect(response.status).toEqual(200);
+      });
+  });
+  it("Check object properties", async () => {
+    await axios
+      .get("http://localhost:8080/api/categories")
```
Comment (dzenetaahmic-abh) [inline test/specs/test.e2e.js]: Can we save this url as global var because we use them in all methods?

File: test/specs/test.e2e.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+          expect(category.name).toExist();
+          expect(category.subCategories).toExist();
+        });
+      });
+  });
+});
+describe("Products API tests", () => {
+  it("Check status code", async () => {
+    await axios
+      .get("http://localhost:8080/api/products")
```
Comment (dzenetaahmic-abh) [inline test/specs/test.e2e.js]: Same comment as above

File: test/specs/test.e2e.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+          let endDate = new Date(product.dateEnd);
+          expect(endDate).toBeGreaterThan(startDate);
+        });
+      });
+  });
+});
+describe("Random product API tests", () => {
+  it("Check status code", async () => {
+    await axios
+      .get("http://localhost:8080/api/products/random")
```
Comment (dzenetaahmic-abh) [inline test/specs/test.e2e.js]: Use global vars for urls

File: test/specs/test.e2e.js [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        expect(data.images).toExist();
+        expect(data.user).toExist();
+      });
+  });
+});
+describe("Product API tests", () => {
+  let idProduct = null;
+
+  beforeAll(() => {
+    const getRandomNumber = () => {
```
Comment (dzenetaahmic-abh) [inline test/specs/test.e2e.js]: We should move this method in separate file, you can create `apiUtils.js` for helper methods

Comment (ajlasisic) [reply test/specs/test.e2e.js]: All improvements are applied in the new commit. @dzenetaahmic-abh

Comment (dzenetaahmic-abh) [reply test/specs/test.e2e.js]: Looks good now, we can merge this pr 👏

**Review by dzenetaahmic-abh — APPROVED** [review]

**Pull request merged**